IS1001Test.py

```diff
@@ -12,12 +12,623 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from GenericTest import GenericTest
+import json
+import socket
+import requests
+from time import sleep
+from urllib.parse import parse_qs
+from OpenSSL import crypto
+
+from GenericTest import GenericTest, NMOSInitException, NMOSTestException
+from Config import AUTH_USERNAME, AUTH_PASSWORD, CERT_TRUST_ROOT_CA, DNS_SD_BROWSE_TIMEOUT, DNS_SD_MODE
+from zeroconf_monkey import ServiceBrowser, Zeroconf
+from MdnsListener import MdnsListener
+
+AUTH_API_KEY = 'auth'
+GRANT_SCOPES = ['is-04', 'is-05']
 
 
 class IS1001Test(GenericTest):
     """
-    Runs IS-10-01-Test
-    """
+    Runs IS-10-01-Test.
+
+    Example "apis" Object:
+    {'auth': {
+      'raml': 'AuthorizationAPI.raml',
+      'version': 'v1.0',
+      'hostname': 'example.co.uk',
+      'port': 80,
+      'spec_branch': 'v1.0-dev',
+      'spec_path': 'cache/is-10',
+      'base_url': 'http://example.co.uk:80',
+      'name': 'Authorization API',
+      'ip': '127.0.1.1',
+      'url': 'http://example.co.uk:80/x-nmos/auth/v1.0/',
+      'spec': <Specification.Specification object at 0x7f5b3b0719b0>
+    }}
+  """
+
     def __init__(self, apis):
-        GenericTest.__init__(self, apis)
+        super(IS1001Test, self).__init__(apis)
+        self.url = self.apis[AUTH_API_KEY]["url"]
+        self.bearer_tokens = []
+        self.client_data = {}
+        self.auth_codes = []
+        self.clients = []  # List of all registered clients for deleting during clean-up
+
+        self.zc = Zeroconf()
+        self.zc_listener = MdnsListener(self.zc)
+
+    def set_up_tests(self):
+        """Add User to Authorization Server"""
+        try:

```

Comment:
If this is just for our implementation we should probably remove it. The README should be updated to state the pre-requisites for running the tests. It may however still be worth checking that the user exists on the auth server before running other tests and raising an exception if so.